index.ts

```diff
@@ -76,7 +76,34 @@ export default function scss(options: CSSPluginOptions = {}): Plugin {
             {
               data: prefix + scss,
               outFile: dest,
-              includePaths
+              includePaths,
+              importer: (
+                url: string,
+                prev: string,
+                done: (
+                  data: { file: string } | { contents: string } | Error | null
+                ) => void
+              ):
+                | { file: string }
+                | { contents: string }
+                | Error
+                | null
+                | void => {
+                let resolved
+
+                if (existsSync(url)) {
+                  resolved = url
+                } else if (existsSync(prev)) {
+                  resolved = prev
+                } else {
+                  const finalUrl = url.startsWith('~')
+                    ? url.replace('~', '')
+                    : url
+                  resolved = require.resolve(finalUrl)
```

Comment:
Presumably you'd need to specify what is making the request as well
```js
require.resolve(finalUrl, { paths: [pathToFile] })
```

Reply:
Valid point. That would *hopefully* remove the need for setting `pnpMode: loose` or doing package overrides. I will try that. I also think I thought of the solution for local resolution. Finally, I still need to clean up the typing a bit and declare some package level types. New commit will be in soon.

Reply:
YES! All tests now pass. Local resolution is fixed. I need to make the commit before I can test with `node_modules` and Yarn PnP.

Reply:
Ok, local resolution is not fixed. Turns out all the tests import sass in JavaScript. There are no local sass import tests. So, that's still broken, but not even necessary. When I was reading the importer documentation, I was under the impression that you had to do all the resolution (as in `libsass`'s native importer was disabled), but I was wrong. I can do the PnP resolution and if that fails, just return `null` and `libsass` will take over.

Reply:
Which tests are you running? Perhaps we could set up Github Actions?

Reply:
@thgh The tests that are build into the repo via `yarn test`. So, to give you an idea on what I am working with at the moment. The following works fine:

- importing a local sass file in JavaScript/TypeScript (in this case Rollup is actually handling the import):
  ```js
  import './styles/styles.scss';
  ```
- importing a sass file from Node in JavaScript/TypeScript (again, in this case Rollup itself is handling the import):
  ```js
  import 'carbon-components/scss/globals/scss/styles.scss';
  ```
- importing a stylesheet from Node in sass (this is handled by my custom importer that uses `require.resolve()` which works on standard `node_modules` projects and Yarn PnP):
  ```scss
  @import '~carbon-components/scss/globals/scss/styles.scss';
  // This will also work because the `~` character doesn't actually do anything
  // I just included support for it because it's kind of a standard in the sass world
  @import 'carbon-components/scss/globals/scss/styles.scss';
  ```

However, the importer breaks one thing and I think I came up with a solution for it:

- importing a local sass file in sass (the resolver resolves this as a file inside Rollup for whatever reason):
  ```scss
  @import './components/Header/Header.scss';
  ```

Currently, I'm only checking if the resolver found a file (i.e. `resolver ? weFoundSomething : weFoundNothing`), but I think the resolver treats local files (i.e. `'./components/Header/Header.scss'`) as if they are part of the package and just returns that directly (i.e. `node_modules/rollup_npm@version/components/Header/Header.scss`), so I think instead I should change my ternary to be:
```ts
return existsSync(resolved) ? done({ file: resolved }) : null
```

and that should fix everything...I hope.

Reply:
I don't know, it's still not working for whatever reason. I really don't want to iterate through the include paths manually or attempt resolve the path of the input file manually and the build a path to the relative file because it shouldn't be necessary. I may have to rely on the `~` character to indicate the file should come from the `node_modules`, and let sass handle everything else...I have one more idea to try, so we'll see.

Reply:
@thgh To fully answer your question by the way. I have a project that I am working on that uses React and Carbon Components (and I've been testing out Yarn PnP for a while), and it does all the different types of sass imports you can do. Importing from node in sass and TypeScript/TSX; as well as importing locally in sass and TypeScript/TSX. So, every time I push a commit, I run over to that project and test it out.